core/cache/model.go

```diff
@@ -65,6 +65,44 @@ func (m *Model) WatchConfig(keys ...string) *ConfigWatcher {
 	return newConfigWatcher(keys, m.hashCache, m.hub, m.topic(modelConfigChange))
 }
 
+func (m *Model) mark() int {
+	var result int
+	m.mu.Lock()
+	for _, app := range m.applications {
+		app.mark()
+		result++
+	}
+	m.mu.Unlock()
+	return result
+}
+
+func (m *Model) sweep() *SweepChecker {
+	checker := &SweepChecker{}
+	m.mu.Lock()
+	for key, app := range m.applications {
+		if checker.Check(app) {
+			delete(m.applications, key)
```

Comment:
I think we need to be a bit more careful than just delete. What if there were watchers on the application?
We should probably process this in the same way a a normal application removal. Similarly for the other aspects of the cache.

Reply:
This is what I'm after, I knew this was too easy!